projects/chaos/src/power_path.c

```diff
@@ -26,6 +26,17 @@ static bool prv_addr_eq(GPIOAddress addr1, GPIOAddress addr2) {
   return ((addr1.port == addr2.port) && (addr1.pin == addr2.pin));
 }
 
+static void prv_send(SoftTimerID timer_id, void *context) {
+  (void)timer_id;
+  PowerPathCfg *cfg = context;
+  PowerPathVCReadings aux = { 0 };
+  power_path_read_source(&cfg->aux_bat, &aux);
```

Comment:
I would suggest using the ADC continuous mode if you aren't already so we don't block in the ADC driver.

Reply:
This is non-blocking. I cache the result every time I do a read already.